pkg/lintcontext/context.go

```diff
@@ -8,13 +8,13 @@ import (
 // ObjectMetadata is metadata about an object.
 type ObjectMetadata struct {
 	FilePath string
-	Raw      []byte
+	Raw      []byte `json:"-"`
 }
 
 // An Object references an object that is loaded from a YAML file.
 type Object struct {
 	Metadata  ObjectMetadata
-	K8sObject k8sutil.Object
+	K8sObject k8sutil.Object `json:"-"` // TODO: find a way to include sufficient information such as entity name and namespace
```

Comment:
We could write a custom `MarshalJSON` function here and pull those out?

Reply:
Good idea. Implemented.

Here's plain text output:

![image](https://user-images.githubusercontent.com/537715/115288103-57a84300-a151-11eb-9cc6-0be717ee5d78.png)

Here's JSON output:

```
╰─➤  bin/linux/kube-linter --format=json lint $HOME/tmp/splunk.yaml | jq
Error: found 6 lint errors
{
  "Reports": [
    {
      "Diagnostic": {
        "Message": "container \"splunk\" does not have a read-only root file system"
      },
      "Check": "no-read-only-root-fs",
      "Remediation": "Set readOnlyRootFilesystem to true in your container's securityContext.",
      "Object": {
        "Metadata": {
          "FilePath": "/home/misha/tmp/splunk.yaml"
        },
        "K8sObject": {
          "Namespace": "splunk",
          "Name": "splunk",
          "GroupVersionKind": "apps/v1, Kind=Deployment"
        }
      }
    },
    {
      "Diagnostic": {
        "Message": "container \"splunk\" is not set to runAsNonRoot"
      },
      "Check": "run-as-non-root",
      "Remediation": "Set runAsUser to a non-zero number, and runAsNonRoot to true, in your pod or container securityContext. See https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for more details.",
      "Object": {
        "Metadata": {
          "FilePath": "/home/misha/tmp/splunk.yaml"
        },
        "K8sObject": {
          "Namespace": "splunk",
          "Name": "splunk",
          "GroupVersionKind": "apps/v1, Kind=Deployment"
        }
      }
    },
    {
      "Diagnostic": {
        "Message": "container \"splunk\" has cpu request 0"
      },
      "Check": "unset-cpu-requirements",
      "Remediation": "Set your container's CPU requests and limits depending on its requirements. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for more details.",
      "Object": {
        "Metadata": {
          "FilePath": "/home/misha/tmp/splunk.yaml"
        },
        "K8sObject": {
          "Namespace": "splunk",
          "Name": "splunk",
          "GroupVersionKind": "apps/v1, Kind=Deployment"
        }
      }
    },
    {
      "Diagnostic": {
        "Message": "container \"splunk\" has cpu limit 0"
      },
      "Check": "unset-cpu-requirements",
      "Remediation": "Set your container's CPU requests and limits depending on its requirements. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for more details.",
      "Object": {
        "Metadata": {
          "FilePath": "/home/misha/tmp/splunk.yaml"
        },
        "K8sObject": {
          "Namespace": "splunk",
          "Name": "splunk",
          "GroupVersionKind": "apps/v1, Kind=Deployment"
        }
      }
    },
    {
      "Diagnostic": {
        "Message": "container \"splunk\" has memory request 0"
      },
      "Check": "unset-memory-requirements",
      "Remediation": "Set your container's memory requests and limits depending on its requirements. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for more details.",
      "Object": {
        "Metadata": {
          "FilePath": "/home/misha/tmp/splunk.yaml"
        },
        "K8sObject": {
          "Namespace": "splunk",
          "Name": "splunk",
          "GroupVersionKind": "apps/v1, Kind=Deployment"
        }
      }
    },
    {
      "Diagnostic": {
        "Message": "container \"splunk\" has memory limit 0"
      },
      "Check": "unset-memory-requirements",
      "Remediation": "Set your container's memory requests and limits depending on its requirements. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for more details.",
      "Object": {
        "Metadata": {
          "FilePath": "/home/misha/tmp/splunk.yaml"
        },
        "K8sObject": {
          "Namespace": "splunk",
          "Name": "splunk",
          "GroupVersionKind": "apps/v1, Kind=Deployment"
        }
      }
    }
  ],
  "Summary": {
    "ChecksStatus": "Failed",
    "CheckTime": "2021-04-19T18:53:57.683205273Z",
    "KubeLinterVersion": "0.2.0-31-g68e42ad8b6-dirty"
  }
}
```